src/components/XdrViewer.js

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import _ from 'lodash';
+import SelectPicker from './FormComponents/SelectPicker';
+import extrapolateFromXdr from '../utilities/extrapolateFromXdr';
+import TreeView from './TreeView';
+import validateTxXdr from '../utilities/validateTxXdr';
+import {updateXdrInput, updateXdrType} from '../actions/xdrViewer';
+
+function XdrViewer(props) {
+  let {dispatch, state} = props;
+
+  let validation = validateTxXdr('');
+  let messageClass = validation.result === 'error' ? 'xdrInput__message__alert' : 'xdrInput__message__success';
+  let message = <p className={messageClass}>{validation.message}</p>
+
+  let xdrNodes, treeView, errorMessage;
+  if (state.input === '') {
+    errorMessage = <p>Enter a base-64 encoded xdr blob to decode.</p>;
+  } else if (state.type === '' || !_.has(xdrTypes, state.type)) {
+    errorMessage = <p>Please select a valid xdr type</p>;
+  } else {
+    try {
```

Comment:
I don't understand what is going on with this block:  Why is `xdrNodes` being declared using the static type of `'TransactionEnvelope'`, but then on the immediate next line you're parsing the XDR again using `state.type`?  Why aren't you using `xdrNodes` in your tree view?